File: Data Structures & Algorithms/majority-element/submission-1.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        ans=0
        pair=dict()
    
        for i in range(len(nums)):
            if nums[i] not in pair:
                pair[nums[i]] =1
            else:
                pair[nums[i]] += 1
        n = int(len(nums)/2)
        for i in pair:
            if(pair[i] >n):
                ans=i
        return ans

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        unordered_map <int, int> freq;
        vector<int> ans;

        for(int i : nums){
            freq[i]++;
        }
        while(k--){
            int max_freq=0;
            int element;
            for(auto it : freq){
                if(it.second>max_freq){
                    max_freq = it.second;
                    element = it.first;
                }
            }
            ans.push_back(element);
            freq.erase(element);
        }
        return ans;
    }
};
